src/ow1/heroes/sombra.opy

```diff
@@ -28,3 +28,34 @@ rule "[sombra.opy]: Cancel invis if hack activated":
     @Condition eventPlayer.isFiringSecondaryFire()
     
     eventPlayer.forceButtonPress(Button.ABILITY_1)
+
+rule "[sombra.opy]: Set hack cooldown time after being damaged while hacking":
+    @Event playerTookDamage
+    @Hero sombra
+    @Condition eventPlayer.isFiringSecondaryFire() == true
+
+    eventPlayer.setAbilityCooldown(Button.SECONDARY_FIRE, OW1_SOMBRA_HACK_COOLDOWN_TIME_AFTER_INTERRUPT)
+
+rule "[sombra.opy]: Set hack duration on heroes":
+    @Event eachPlayer
+    @Condition eventPlayer.hasStatusEffect(Status.HACKED) == true
+
+    wait(OW1_SOMBRA_HACK_DURATION_ON_ENEMIES)
+    if eventPlayer.isAlive():
+        respawnSameState()
+
+rule "[sombra.opy]: Disable abilities on hacked enemies":
+    @Event eachPlayer
+    @Condition eventPlayer.hasStatusEffect(Status.HACKED) == true
+
+    eventPlayer.setAbility1Enabled(false)
+    eventPlayer.setAbility2Enabled(false)
+    eventPlayer.disallowButton(Button.ULTIMATE)
+    # Disables secondary fire on heroes that have a separate ability as secondary fire. Weapons should not be affected.
+    if not eventPlayer.getCurrentHero() in [Hero.ANA, Hero.ASHE, Hero.BAPTISTE, Hero.MCCREE, Hero.GENJI, Hero.JUNKRAT, Hero.MEI, Hero.ROADHOG, Hero.SYMMETRA, Hero.TORBJORN, Hero.TRACER, Hero.WIDOWMAKER, Hero.ZARYA]:
+        eventPlayer.setSecondaryFireEnabled(false)
+    wait(OW1_SOMBRA_HACK_DURATION_ON_ENEMIES)
+    eventPlayer.setSecondaryFireEnabled(true)
```

Comment:
I think for example for monkey it will reactivate his secondary fire which is desactivated at the init of the hero : https://github.com/MaxwellJung/ow1_emulator/blob/staging/src/ow1/heroes/winston.opy#L23

Reply:
Correct. Will fix now.